source/extensions/filters/network/common/factory_base.h

```diff
@@ -56,7 +58,10 @@ class FactoryBase : public Server::Configuration::NamedNetworkFilterConfigFactor
   virtual Upstream::ProtocolOptionsConfigConstSharedPtr
   createProtocolOptionsTyped(const ProtocolOptionsProto&,
                              Server::Configuration::ProtocolOptionsFactoryContext&) {
-    throw EnvoyException(fmt::format("filter {} does not support protocol options", name_));
+    ExceptionUtil::throwEnvoyException(
+        fmt::format("filter {} does not support protocol options", name_));
+    // to satisfy compiler and should never reach here.
+    return nullptr;
```

Comment:
I think you can annotate the `throwEnvoyException()` with the `ABSL_ATTRIBUTE_NORETURN` attribute to prevent compiler from complaining about missing return statements.

Reply:
yes, I actually did but forgot to remove the artifact here.